newdle/client/src/components/summary/SummaryPage.js

```diff
@@ -229,88 +212,130 @@ export default function SummaryPage() {
+              >
+                <Dropdown.Menu>
+                  <Dropdown.Item as={Link} text={t`Participants`} to={`${editUrl}/participants`} />
+                  <Dropdown.Item as={Link} text={t`Timeslots`} to={editUrl} />
+                  <Dropdown.Item as={Link} text={t`Options`} to={`${editUrl}/options`} />
+                  <Dropdown.Item
+                    text={t`Delete newdle`}
+                    onClick={() => setDeletionModalOpen(true)}
+                  />
+                </Dropdown.Menu>
+              </Dropdown>
+            </div>
+          )}
+          <DeleteModal open={deletionModalOpen} setOpen={setDeletionModalOpen} />
         </>
       )}
     </Container>
   );
 }
+
+export function DeleteModal({open, setOpen}) {
+  const [sendDeletionMail, setSendDeletionMail] = useState(false);
+  const [comment, setComment] = useState('');
+  const newdle = useSelector(getNewdle);
+  const hasParticipantsWithEmail = useSelector(newdleHasParticipantsWithEmail);
+  const hasParticipantsWithoutEmail = useSelector(newdleHasParticipantsWithoutEmail);
+  const participantsWithoutEmail = useSelector(newdleParticipantsWithoutEmail);
+  const dispatch = useDispatch();
+  const [_deleteNewdle, deleting] = client.useBackendLazy(client.deleteNewdle);
+  const [_sendDeletionEmails, deletionMailSending] = client.useBackendLazy(
+    client.sendDeletionEmails
+  );
+
+  const handleClose = useCallback(() => {
+    setOpen(false);
+  }, [setOpen]);
+
+  const handleConfirm = useCallback(async () => {
+    if (sendDeletionMail) {
+      _sendDeletionEmails(newdle.code, comment.trim());
+    }
+    const deletedNewdle = await _deleteNewdle(newdle.code);
+    if (deletedNewdle) {
+      dispatch(updateNewdle(deletedNewdle));
+    }
+    setOpen(false);
+  }, [
+    _deleteNewdle,
+    _sendDeletionEmails,
+    comment,
+    dispatch,
+    newdle.code,
+    sendDeletionMail,
+    setOpen,
+  ]);
+
+  return (
+    <Modal onClose={handleClose} size="small" closeIcon open={open}>
+      <Modal.Header>
+        <Trans>Delete this newdle</Trans>
+      </Modal.Header>
+      <Modal.Content>
+        <Trans>
+          Do you really wish to delete this newdle? Please be aware that{' '}
+          <strong>this action cannot be undone.</strong>
+        </Trans>
+        {hasParticipantsWithEmail && (
+          <div className={styles['mail-checkbox']}>
+            <Checkbox
+              checked={sendDeletionMail}
+              onChange={() => setSendDeletionMail(!sendDeletionMail)}
+              label="Notify the participants via e-mail"
+            />
+            {sendDeletionMail && (
+              <Input
+                type="text"
+                placeholder={t`Leave a comment (optional)`}
+                className={styles['deletion-comment']}
+                value={comment}
+                onChange={(__, {value}) => setComment(value)}
+              />
+            )}
+            {hasParticipantsWithoutEmail && (
```

Comment:
Unrelated, but shouldn't we show this only if  `hasParticipantsWithoutEmail && sendDeletionMail`? Showing this warning when you didn't select to notify people is a bit pointless..

Reply:
I agree